### PR TITLE
Task generate models

### PR DESCRIPTION
**Pull request by dmuriel**
This PR adds a new generator for oxpecker, by typing:
```
ox generate model user
```
ox will create 2 files named "user.go" and "user_test.go", and by default will include the ID, CreatedAt and UpdatedAt attributes.

If you type:
```
ox generate model user salary:int title
```
ox will generate the same 2 files with the defaults attributes and will also include the salary and title attributes. If the attribute does not have a type, it will return "string" by default.

### CODE REVIEW

File: tools/model/templates.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+`
+
+var modelTestTemplate string = `package models
+
+func (ms *ModelSuite) Test_{{ properize .Name }}() {
+	ms.Fail("This test needs to be implemented!")
+}`
+
+var templateFuncs = template.FuncMap{
```
Comment (paganotoni) [inline tools/model/templates.go:41]: Love this approach Danny!

File: tools/model/templates.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	{{ pascalize $attr.Name }}	{{$attr.GoType }} ` + "`" + `json:"{{ underscore $attr.Name }}" db:"{{ underscore $attr.Name }}"` + "`" + `
+	{{- end }}
+}
+
+// {{ pluralize .Name }} array model struct of {{ properize .Name }}
+type {{ pluralize .Name }} []{{ properize .Name }}
+
+// String converts the struct into a string value
+func ({{ .Char }} {{ properize .Name }}) String() string {
+	js, _ := json.Marshal({{ .Char }})
```
Comment (paganotoni) [inline tools/model/templates.go:29]: @dmuriel I think we should handle error here. (This is a common issue generated models have)

File: plugins.go [hunk 3]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/wawandco/oxplugins/tools/envy"
 	"github.com/wawandco/oxplugins/tools/flect"
 	"github.com/wawandco/oxplugins/tools/grift"
+	"github.com/wawandco/oxplugins/tools/model"
```
Comment (paganotoni) [inline plugins.go:16]: @dmuriel could we put this package inside the buffalo one? I think that will be good.
```suggestion
	"github.com/wawandco/oxplugins/tools/buffalo/model"
```

**Review by paganotoni — COMMENTED** [review]
Looking great @dmuriel !

**Review by paganotoni — CHANGES_REQUESTED** [review]
Just a small set o tweaks.

File: tools/model/templates.go (hunk 2 above)
Comment (dmuriel) [reply tools/model/templates.go]: I just add a "log.Fatal(err)" on the template.

File: tools/model/templates.go (hunk 1 above)
Comment (dmuriel) [reply tools/model/templates.go]: Thanks :)

**Comment by dmuriel** [comment]
@paganotoni I make the changes you requested.

File: tools/buffalo/model/templates.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+// {{ pluralize .Name }} array model struct of {{ properize .Name }}
+type {{ pluralize .Name }} []{{ properize .Name }}
+
+// String converts the struct into a string value
+func ({{ .Char }} {{ properize .Name }}) String() string {
+	js, err := json.Marshal({{ .Char }})
+	if err != nil {
+		log.Fatal(err)
```
Comment (paganotoni) [inline tools/buffalo/model/templates.go:31]: ^^ this one is not good :D. Lets throw JSON and do:
```
return fmt.Sprintf("%+v\n", {{.Char}})
```

**Review by paganotoni — CHANGES_REQUESTED** [review]

File: tools/buffalo/model/templates.go (hunk 4 above)
Comment (dmuriel) [reply tools/buffalo/model/templates.go]: Got it. Thanks for this, i'll keep it on mind for future references.

**Review by paganotoni — APPROVED** [review]

**Pull request merged**